src/main/java/codeu/model/store/persistence/PersistentDataStore.java

```diff
@@ -172,8 +178,13 @@ public PersistentDataStore() {
         UUID ownerUuid = UUID.fromString((String) entity.getProperty("owner_uuid"));
         String title = (String) entity.getProperty("title");
         Instant creationTime = Instant.parse((String) entity.getProperty("creation_time"));
-        byte[] contentBytes = ((String) entity.getProperty("content")).getBytes(StandardCharsets.UTF_8);
-        Media singleMedia = new Media(uuid, ownerUuid, title, creationTime, contentBytes);
+        String contentString = ((Text) entity.getProperty("content")).getValue();
+        byte[] contentBytes = new BigInteger(contentString, 16).toByteArray();
+        InputStream in = new ByteArrayInputStream(contentBytes);
+        BufferedImage content = ImageIO.read(in);
```

Comment:
Shouldn't you check the content type to image first, then do
BufferedImage content = ImageIO.read(in);
?